homeassistant/components/vera/__init__.py

```diff
@@ -159,43 +171,52 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 def map_vera_device(vera_device, remap):
     """Map vera classes to Home Assistant types."""
 
-    if isinstance(vera_device, veraApi.VeraDimmer):
-        return "light"
-    if isinstance(vera_device, veraApi.VeraBinarySensor):
-        return "binary_sensor"
-    if isinstance(vera_device, veraApi.VeraSensor):
-        return "sensor"
-    if isinstance(vera_device, veraApi.VeraArmableDevice):
-        return "switch"
-    if isinstance(vera_device, veraApi.VeraLock):
-        return "lock"
-    if isinstance(vera_device, veraApi.VeraThermostat):
-        return "climate"
-    if isinstance(vera_device, veraApi.VeraCurtain):
-        return "cover"
-    if isinstance(vera_device, veraApi.VeraSceneController):
-        return "sensor"
-    if isinstance(vera_device, veraApi.VeraSwitch):
-        if vera_device.device_id in remap:
+    type_map = {
+        veraApi.VeraDimmer: "light",
+        veraApi.VeraBinarySensor: "binary_sensor",
+        veraApi.VeraSensor: "sensor",
+        veraApi.VeraArmableDevice: "switch",
+        veraApi.VeraLock: "lock",
+        veraApi.VeraThermostat: "climate",
+        veraApi.VeraCurtain: "cover",
+        veraApi.VeraSceneController: "sensor",
+        veraApi.VeraSwitch: "switch",
+    }
+
+    def map_special_case(instance_class: Type, entity_type: str) -> str:
+        if instance_class is veraApi.VeraSwitch and vera_device.device_id in remap:
             return "light"
-        return "switch"
-    return None
+        return entity_type
+
+    return next(
+        iter(
+            map_special_case(instance_class, entity_type)
+            for instance_class, entity_type in type_map.items()
+            if isinstance(vera_device, instance_class)
+        ),
+        None,
+    )
 
 
 class VeraDevice(Entity):
     """Representation of a Vera device entity."""
 
-    def __init__(self, vera_device, controller):
+    def __init__(self, vera_device, controller_data: ControllerData):
```

Comment:
I missed this before I merged it, but usually we type all arguments or no arguments.

We should type `vera_device` here or remove `ControllerData`

Reply:
Vera typings are cleaned up in #40143